### ISSUE TITLE
Fix dependency resolution for PCL assemblies

### ISSUE THREAD

**Issue by paulbatum** [Azure/azure-webjobs-sdk-script#178]
I dont know if this is a script host bug or a project oxford bug package bug or something else but I found this behavior strange.

Steps to reproduce:

Add a project.json that references project oxford computer vision api:

```
{
  "frameworks": {
    "net46":{
      "dependencies": {
        "Microsoft.ProjectOxford.Vision": "0.5.0.1" 
      }
    }
  }
}
```

Attempt to use the SDK from an async function:

```
#r "Microsoft.WindowsAzure.Storage"

using System;
using System.Threading.Tasks;
using Microsoft.WindowsAzure.Storage.Blob;
using Microsoft.ProjectOxford.Vision;

public static async Task Run(ICloudBlob myBlob, TraceWriter log, IAsyncCollector<object> document)
{
    var visionClient = new VisionServiceClient("asdf");

    var result = await visionClient.RecognizeTextAsync(myBlob.Uri.ToString(), "en"); 

    var words = from r in result.Regions
        from l in r.Lines
        from w in l.Words
        select w.Text;

    var output = string.Join(" ", words.ToArray());
    log.Verbose($"Recognized words: " + output);

    myBlob.Metadata["ocr"] = output;
    await myBlob.SetMetadataAsync();

    var doc = new {
        Id = myBlob.Uri,
        Type = "ImageDocument",
        Words = words
    };

    await document.AddAsync(doc);
} 
```

Expected result: compiles successfully.
Actual result: 

```
2016-03-28T02:44:04.588 Script for function 'ocrtest' changed. Reloading.
2016-03-28T02:44:04.588 Compiling function script.
2016-03-28T02:44:04.713 (12,37): error CS0012: The type 'Object' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
2016-03-28T02:44:04.713 (12,24): error CS0012: The type 'Task<>' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Threading.Tasks, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
2016-03-28T02:44:04.713 Compilation failed.
```

The workaround is to add the following but why is this necessary?

```
#r "System.Runtime"
#r "System.Threading.Tasks"
#r "System.IO"
```

**Comment by fabiocav** [comment]
Strange indeed. This shouldn't be required. I'm suspecting something with the version it's requesting, but I'll take a look at this in the morning.

**Comment by christopheranderson** [comment]
Still need a root cause on this in order to triage.

**Comment by mathewc** [comment]
@fabiocav Wondering if all the work that has happened in assembly referencing since this was logged has addressed this?

**Comment by mathewc** [comment]
@paulbatum A lot of improvements have been made since this issue was first logged. Can you tell us if it still repros?

**Comment by sxmichael** [comment]
Bump, just hit this issue. When using custom project.json file, received same compilation errors of Object and Task<> undefined. Workaround above solved the problem.
```
#r "System.Runtime"
#r "System.Threading.Tasks"
```

My project.json:
```
{
  "dependencies": {
    "Microsoft.Azure.Management.Network": "8.0.0-preview",
    "Microsoft.Azure.Management.ResourceManager": "1.4.0-preview",
    "Microsoft.IdentityModel.Clients.ActiveDirectory": "3.13.7",
    "Microsoft.Rest.ClientRuntime": "2.1.0",
    "Microsoft.Rest.ClientRuntime.Azure": "3.1.0"
  },
  "frameworks": {
    "net46": {
    }
  }
}
```

**Comment by fabiocav** [comment]
@sxmichael , this is indeed a known issue/limitation involving PCL scenarios. ( SO answer on rlthis here http://stackoverflow.com/questions/40192646/azure-function-consuming-onedrive-api-returns-compile-errors/40195875#40195875)

Skipping the details of the underlying root cause, this is a nuance when referencing PCLs and we do have some options planned to remove the requirement, but your workaround is the correct approach at the moment.

I'll reopen this so it is tracked

**Comment by fabiocav** [comment]
Most scenarios impacted by this should be addressed with #1049, which will be in our next release.
